File: Laboratorios/lab_10/labcsv-completo-empresa-departamento-funcionarios/empresa.cpp
#include "empresa.h"

Departamento &Empresa::createDepartamento(int codigo, const string &nome)
{
    Departamento d(codigo,nome);
    departamentos.insert({codigo,d});
    return departamentos.at(codigo);
}

bool Empresa::containsDepartamento(int codigo) const
{
    return departamentos.find(codigo)!=departamentos.end();
}

Departamento &Empresa::findDepartamento(int codigo)
{
    return departamentos.at(codigo);
}

const map<int, Departamento> &Empresa::getDepartamentos() const
{
    return this->departamentos;
}

File: Laboratorios/lab_2/aula-04-empresa/Empresa.java
import java.util.LinkedList;

public class Empresa
{
    private LinkedList<Departamento> departamentos = new LinkedList<Departamento>();
    private String nome;
    
    public Empresa(String nome) {
        this.nome = nome;
    }
    public String getNome() {
        return nome;
    }
    public void setNome(String nome) {
        this.nome = nome;
    }

    public void adicionaDepartamento(Departamento d)
    {
        departamentos.add(d);
    }
    public void removeDepartamento(Departamento d)
    {
        departamentos.remove(d);
    }
    public LinkedList<Departamento> getDepartamentos()
    {
        return new LinkedList<Departamento>(departamentos);
    }

    public void aumentaSalarioFuncionarios(float porcentagem)
    {
        for(Departamento d : departamentos)
        {
            d.aumentaSalarioFuncionarios(porcentagem);
        }
    }

    public float getSalarioMedio()
    {
        int i;
        float soma=0.0f;
        i=0;
        for(Departamento d : departamentos)
        {
            for(Funcionario f : d.getFuncionarios())
            {
                soma+=f.getSalario();
                i++;
            }
        }
        return soma/i;
    }

}